kafkakewl-metrics/src/main/scala/com/mwam/kafkakewl/metrics/services/HighOffsetMetricsExposer.scala

```diff
@@ -93,29 +130,40 @@ case class MetricMap(metricMap: Map[String, (Map[Int, (Metric.Counter[Long], Lon
   }
 }
 
+class HighOffsetMetricsExposer
+
 object HighOffsetMetricsExposer {
-  def startPublishing(
-      topicInfos: ZStream[MetricExposerConfig, Nothing, KafkaTopicPartitionInfoChanges]
-  ): URIO[MetricExposerConfig, Fiber.Runtime[Nothing, Unit]] =
-    for {
-      config <- ZIO.service[MetricExposerConfig]
-      fibre <- topicInfos
-        .runFoldZIO(MetricMap(Map.empty)) { (metricMap: MetricMap, topicPartitionInfoChanges: KafkaTopicPartitionInfoChanges) =>
-          val (newMetricMap, metricUpdates) = metricMap.updateWithTopicParitionChanges(topicPartitionInfoChanges, config)
-          ZIO.foreachDiscard(metricUpdates)((metric, increase) => metric.update(increase)).as(newMetricMap)
-        }
-        .unit
-        .fork
-    } yield fibre
+  def live: ZLayer[KafkaTopicInfoSource & TopicsConfig, Nothing, HighOffsetMetricsExposer] =
+    ZLayer.scoped {
+      for {
+        topicInfoService <- ZIO.service[KafkaTopicInfoSource]
+        topicInfoStream = topicInfoService.subscribeStream()
+        config <- ZIO.service[TopicsConfig]
+
+        offsetMetricFibre <- topicInfoStream
+          .runFoldZIO(MetricMap(Map.empty)) { (metricMap: MetricMap, topicPartitionInfoChanges: KafkaTopicPartitionInfoChanges) =>
+            val (newMetricMap, metricUpdates) = metricMap.updateWithTopicPartitionChanges(topicPartitionInfoChanges, config)
+            ZIO.foreachDiscard(metricUpdates)(_.applyChanges()).as(newMetricMap)
+          }
+          .unit
+          .fork
+        _ <- ZIO.addFinalizer(offsetMetricFibre.interrupt)
+      } yield HighOffsetMetricsExposer()
+    }
 }
 
-case class MetricExposerConfig(perPartitionMetrics: Boolean = true)
+final case class TopicsConfig(perPartitionHighOffsetMetrics: Boolean = false, metricNamePrefix: String = "kafkakewl_vnext_metrics_") {
```

Comment:
I think for now this will do, but later we'll have to revisit configuring the metrics name prefix, because there are other metrics elsewhere in the code already, and they all should have the same prefix. So it's not really `TopicsConfig` specific. But it can be fixed as a separate change.

Reply:
Actually maybe the default could be just `kafkakewl_vnext_`, no need the `metrics` in the metrics name.